telemetry/packages/fake/src/sensorClasses/index.ts

```diff
@@ -0,0 +1,37 @@
+// Individual sensor classes
+import { Motion } from './motion';
+import { Keyence } from './keyence';
+import { Pressure } from './pressure';
+import { Temperature } from './temperature';
+import { Resistance } from './resistance';
+import { Magnetism } from './magnetism';
+import { Levitation } from './levitation';
+
+type SensorType =
+  | typeof Motion
+  | typeof Keyence
+  | typeof Pressure
+  | typeof Temperature
+  | typeof Resistance
+  | typeof Magnetism
+  | typeof Levitation;
+
+// Package all sensors into one object to be iterated through easily in SensorManager
+export type sensorObj = {
+  [sensor: string]: SensorType;
+};
+
+// Instance type for sensor classes
+export type SensorInstance<T extends new (...args: any[]) => any> =
```

Comment:
this is a weird type def...

Reply:
We decided to leave it for now, necessary for type recognition. The type gives a structure to items in the array of sensor instances, so TS recognises that .getData, .getRandomData and the shared static attributes are defined on any given sensor instance.